src/utils/env.ts

```diff
@@ -26,6 +26,28 @@ const [defaultEnv] = environments.filter(env => env.region === DEFAULT_ENV_REGIO
 
 export const DEFAULT_PC_ENV = buildPcEnv(defaultEnv.publicDomainName);
 
+function findPcEnvironment(location: URL, targetEnv: string, parseEnvironment: EnvironmentParser): PcEnv|null {
+    const parsedEnv = parseEnvironment(location.origin, {});
+    if (parsedEnv && parsedEnv.name === targetEnv) {
+        return {
+            pcEnvTld: parsedEnv.publicDomainName,
+            pcAppOrigin: location.origin
+        };
+    } else {
+        for (const environment of __GC_DEV_EXTRA_ENVS__) {
+            const publicDomains = [environment.publicDomainName, ...(environment.publicDomainAliases || [])];
+            const matchingDomain = publicDomains.find((p) => location.hostname.endsWith(p));
```

Comment:
small nit here ... the env domains don't have a leading `.` so we have to add it here.

`location.hostname === p || location.hostname.endsWith(`.${p}`)`

otherwise, `apps.evilmypurecloud.com` would match `mypurecloud.com`

just lessons learned from a lot of unit testing parsing domains.  Granted, this is almost entirely for dev/testing, but, we might as well be safe.